halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java

```diff
@@ -570,6 +604,36 @@ default RunningServiceDetails getRunningServiceDetails(AccountDeploymentDetails<
     return res;
   }
 
+  @Override
+  default <S> S connectToInstance(AccountDeploymentDetails<KubernetesAccount> details, SpinnakerRuntimeSettings runtimeSettings, SpinnakerService<S> sidecar, String instanceId) {
+    ServiceSettings settings = runtimeSettings.getServiceSettings(sidecar);
+    String namespace = getNamespace(settings);
+    int localPort = SocketUtils.findAvailableTcpPort();
+    int targetPort = settings.getPort();
+    List<String> command = KubernetesProviderUtils.kubectlPortForwardCommand(details,
+        namespace,
+        instanceId,
+        targetPort,
+        localPort);
+    JobRequest request = new JobRequest().setTokenizedCommand(command);
+    String jobId = getJobExecutor().startJob(request);
+
+    // Wait for the proxy to spin up.
+    DaemonTaskHandler.safeSleep(TimeUnit.SECONDS.toMillis(5));
```

Comment:
Tricky thing is that there is nothing clean to poll on (no clear success condition). If we try to open a connection to the local port, it's possible the container isn't accepting requests yet, or that there is already something else bound to the local port (false negative and false positive response respectively).